overlord/snapstate/snapstate_test.go

```diff
@@ -571,6 +571,19 @@ func checkIsAutoRefresh(c *C, tasks []*state.Task, expected bool) {
 	c.Fatalf("cannot find download-snap task in %v", tasks)
 }
 
+func checkIsContinuedAutoRefresh(c *C, tasks []*state.Task, expected bool) {
+	for _, t := range tasks {
+		if t.Kind() == "download-snap" {
```

Comment:
I would prefer to call `TaskSnapSetup` with any task in the change. That helper looks for the SnapSetup or for an associated task that has it. IMO, it would be simpler and more robust (if something changes and the change has a "prepare-snap" task instead of a "download-snap", this would break). But it's not a big deal, so feel free to ignore this

Reply:
I mirrored the pre-existing `checkIsAutoRefresh` tbh, but happy to tweak both in a followup, the concern is absolutely justified.

Reply:
I assume you mean something like:
```go
index bdaa41de2c..3a73f01c14 100644
--- a/overlord/snapstate/snapstate_test.go
+++ b/overlord/snapstate/snapstate_test.go
@@ -572,16 +572,11 @@ func checkIsAutoRefresh(c *C, tasks []*state.Task, expected bool) {
 }
 
 func checkIsContinuedAutoRefresh(c *C, tasks []*state.Task, expected bool) {
-       for _, t := range tasks {
-               if t.Kind() == "download-snap" {
-                       var snapsup snapstate.SnapSetup
-                       err := t.Get("snap-setup", &snapsup)
-                       c.Assert(err, IsNil)
-                       c.Check(snapsup.IsContinuedAutoRefresh, Equals, expected)
-                       return
-               }
-       }
-       c.Fatalf("cannot find download-snap task in %v", tasks)
+       var snapsup snapstate.SnapSetup
+       err := tasks[0].Get("snap-setup", &snapsup)
+       c.Assert(err, IsNil)
+       c.Check(snapsup.IsContinuedAutoRefresh, Equals, expected)
+       return
 }
 
 func (s *snapmgrTestSuite) TestLastIndexFindsLast(c *C) {
```

Reply:
Yes, that was exactly it. It's nothing major, just a small detail